kernel/src/snapshot.rs

```diff
@@ -248,6 +251,10 @@ impl Snapshot {
     pub fn column_mapping_mode(&self) -> ColumnMappingMode {
         self.column_mapping_mode
     }
+    /// Create a [`ScanBuilder`] for this `Snapshot`.
+    pub fn scan_builder(&self) -> ScanBuilder {
+        ScanBuilder::new(self.clone())
+    }
```

Comment:
I actually think we shouldn't `#[derive(Clone)]` and then clone ourselves. That's somewhat surprising to a user. Instead, let's implement it for `Arc`, and add an `into_scan_builder()` that consumes the snapshot:

```suggestion
    pub fn scan_builder(self: Arc<Self>) -> ScanBuilder {
        ScanBuilder::new(self)
    }
    
    pub fn into_scan_builder(self) -> ScanBuilder {
        ScanBuilder::new(self)
    }
```

You'll need to import `Arc` for this to work. Also please remove the `derives` for `Clone`

Reply:
ok added scan_builder, into_scan_builder